src/Datasource/EntityInterface.php

```diff
@@ -95,9 +95,14 @@ public function visibleProperties();
      * Returns an array with all the properties that have been set
      * to this entity
      *
+     * ### Options
+     *
+     * * `includeHidden` Include hidden properties in array
+     *
+     * @param array $options List of options
      * @return array
      */
-    public function toArray();
+    public function toArray($options = []);
```

Comment:
@markstory so we keep the option and allProperties() in default implementation only ?